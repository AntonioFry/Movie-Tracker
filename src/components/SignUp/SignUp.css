
.sign-in-form{
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin: auto;
}

.sign-up{
  background-color: black;
  border-color: red;
  color: red;
  height: 30px;
  margin: 10px;
  width: 175px;
}

.sign-up-btn {
  background-color: black;
  border-color: red;
  border-radius: 20px;
  color: red;
  font-size: 1.5rem;
  height: 35px;
  margin: 10px;
  width: 150px;
}

::placeholder {
  color: red;
  font-size: 15px;
}
