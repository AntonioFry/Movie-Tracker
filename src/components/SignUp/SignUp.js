import React from 'react';

export const SignUp = ({name, email, password, handleChange, handleSubmit}) => {

  return (
    <section>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          value={name}
          name="name"
          placeholder='Name'
          onChange={handleChange}
        />
        <input
          type="text"
          value={email}
          name="email"
          placeholder='example@example.com'
          onChange={handleChange}
        />
        <input
          type="text"
          value={password}
          name="password"
          placeholder='Password'
          onChange={handleChange}
        />
      <button>Sign Up</button>
    </form>
  </section>
  )
};
