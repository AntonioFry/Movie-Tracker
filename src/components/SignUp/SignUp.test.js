import React from 'react';
import { shallow } from 'enzyme';
import { SignUp, mapStateToProps } from './SignUp'

describe('SignUp', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <SignUp 
      name = 'Andre 3000'
      email='andre@outkast.com'
      password='imBetterThanBIGBoi321'
      handleAdd= {jest.fn()}
      handleAddChange= {jest.fn()}
      />
    )
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return a string with an error', () => {
      const mockState = {
        error: 'You fucked up',
        filter: 'SHOW_ERROR'
      }
      const expected = {
        error: 'You fucked up'
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    });
  }); 
});