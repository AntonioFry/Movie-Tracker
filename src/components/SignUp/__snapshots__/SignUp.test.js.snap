// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match snapshot 1`] = `
<section>
  <form
    onSubmit={[MockFunction]}
  >
    <input
      name="name"
      onChange={[MockFunction]}
      placeholder="Name"
      type="text"
      value="Andre 3000"
    />
    <input
      name="email"
      onChange={[MockFunction]}
      placeholder="example@example.com"
      type="text"
      value="andre@outkast.com"
    />
    <input
      name="password"
      onChange={[MockFunction]}
      placeholder="Password"
      type="text"
      value="imBetterThanBIGBoi321"
    />
    <button>
      Sign Up
    </button>
  </form>
</section>
`;
